Tests/AEPOptimizeTests/FunctionalTests/OptimizeFunctionalTests.swift

```diff
@@ -847,6 +847,200 @@ class OptimizeFunctionalTests: XCTestCase {
+                ]
+            ]
+        )
+
+        let getEvent = Event(
+            name: "Optimize Get Propositions Request",
+            type: "com.adobe.eventType.optimize",
+            source: "com.adobe.eventSource.requestContent",
+            data: [
+                "requesttype": "getpropositions",
+                "decisionscopes": [
+                    ["name": decisionScopeA.name]
+                ]
+            ]
+        )
+
+        mockRuntime.simulateSharedState(for: ("com.adobe.module.configuration", updateEvent),
+                                        data: ([
+                                            "edge.configId": "ffffffff-ffff-ffff-ffff-ffffffffffff"] as [String: Any], .set))
+
+        /// Simulating update & get events
+        mockRuntime.simulateComingEvents(updateEvent)
+        mockRuntime.simulateComingEvents(getEvent)
+
+        XCTAssertTrue(mockRuntime.dispatchedEvents.isEmpty)
+
+        let optimizeContentComplete = Event(
+            name: "Optimize Update Propositions Complete",
+            type: "com.adobe.eventType.optimize",
+            source: "com.adobe.eventSource.contentComplete",
+            data: [
+                "completedUpdateRequestForEventId": updateEvent.id.uuidString,
+                "propositions" : [updatedPropositionJSON]
+            ]
+        )
+        
+        mockRuntime.simulateComingEvents(optimizeContentComplete)
+        optimize.cachedPropositions[decisionScopeA] = updatedPropositionForScopeA
+        
+        /// After the update is complete, the get event should now dispatch
+        let expectation = XCTestExpectation(description: "Get propositions request should now dispatch response event after update completion.")
+        
+        mockRuntime.onEventDispatch = { event in
+            expectation.fulfill()
+        }
+        
+        wait(for: [expectation], timeout: 10)
+        XCTAssertEqual(mockRuntime.dispatchedEvents.count, 1)
+        
+        let dispatchedEvent = mockRuntime.dispatchedEvents.first
+        XCTAssertEqual(dispatchedEvent?.type, "com.adobe.eventType.optimize")
+        XCTAssertEqual(dispatchedEvent?.source, "com.adobe.eventSource.responseContent")
+
+        /// Validate that the Get proposition response contains the updated proposition
+        guard let propositionsDictionary: [DecisionScope: OptimizeProposition] = dispatchedEvent?.getTypedData(for: "propositions") else {
+            XCTFail("Propositions dictionary should be valid.")
+            return
+        }
+        XCTAssertEqual(propositionsDictionary[decisionScopeA]?.id, updatedPropositionForScopeA.id)
+        XCTAssertEqual("BBBBBBBB-BBBB-BBBB-BBBB-BBBBBBBBBBBB", propositionsDictionary[decisionScopeA]?.id)
+        XCTAssertEqual("eyJhY3Rpdml0eUlkIjoieGNvcmU6b2ZmZXItYWN0aXZpdHk6MTExMTExMTExMTExMTExMSIsInBsYWNlbWVudElkIjoieGNvcmU6b2ZmZXItcGxhY2VtZW50OjExMTExMTExMTExMTExMTEifQ==", propositionsDictionary[decisionScopeA]?.scope)
+    }
+
```

Comment:
can you also add tests for
1. some requested propositions are in cache, some are not -> verify that the getPropositions event gets queued?
2. updateProposition(A), updatePropositions(B), updatePropositions(A), getPropositions(A) -> verify the get is executed after all updates?

Reply:
Sure, I have a new test case for this scenario.